src/main/java/com/specialwarriors/conal/common/auth/oauth/CustomOAuth2UserService.java

```diff
@@ -0,0 +1,49 @@
+package com.specialwarriors.conal.common.auth.oauth;
+
+import com.specialwarriors.conal.user.domain.User;
+import com.specialwarriors.conal.user.repository.UserRepository;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomOAuth2UserService implements OAuth2UserService<OAuth2UserRequest, OAuth2User> {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
+        OAuth2UserService<OAuth2UserRequest, OAuth2User> delegate =
+                new DefaultOAuth2UserService();
+        OAuth2User oauth2User = delegate.loadUser(userRequest);
+
+        // GitHub 사용자 정보 추출
+        Number idRaw = oauth2User.getAttribute("id");
+        Long githubId = idRaw.longValue();
```

Comment:
```suggestion
        long githubId = oauth2User.getAttribute("id").longValue();
```
두 라인의 코드를 한 줄로 작성할 수 있을 것 같이 보이는데 나누어 작성하신 이유가 있을까요? 또한 기본 타입으로 다룰 수 있을 것 같은데 굳이 래퍼 타입으로 값을 정의한 의도가 궁금합니다.

Reply:
저도 한 줄로 쓰려 했었는데 바로 longValue()를 붙이면 에러가 발생해서 따로 나눠서 작성했습니다.